management/virtualmachine/client.go

```diff
@@ -648,8 +648,8 @@ func (self VirtualMachineClient) createServiceCertDeploymentConf(certPath string
 	return certConfig, nil
 }
 
-func (self VirtualMachineClient) createSshConfig(certPath, userName string) (SSH, error) {
-	sshConfig := SSH{}
+func (self VirtualMachineClient) createSshConfig(certPath, userName string) (*SSH, error) {
```

Comment:
Totally agree and yes, they are only here to make xml.Marshall work properly... :-(

Sent from my Windows Phone
________________________________
From: Ahmet Alp Balkan
Sent: 3/14/2015 2:38 PM
To: MSOpenTech/azure-sdk-for-go
Cc: Paul Meyer
Subject: Re: [azure-sdk-for-go] Update virtual machine entities to latest spec (#64)


In management/virtualmachine/client.go<https://github.com/MSOpenTech/azure-sdk-for-go/pull/64#discussion_r26442000>:

> @@ -648,8 +648,8 @@ func (self VirtualMachineClient) createServiceCertDeploymentConf(certPath string
>       return certConfig, nil
>  }
>
> -func (self VirtualMachineClient) createSshConfig(certPath, userName string) (SSH, error) {
> -     sshConfig := SSH{}
> +func (self VirtualMachineClient) createSshConfig(certPath, userName string) (*SSH, error) {


no big deal. having all these pointers is just annoying. I can understand if we re using them for omitting things in XML marshaling but otherwise we shouldnt be doing any pointers, ideally. right?

-
Reply to this email directly or view it on GitHub<https://github.com/MSOpenTech/azure-sdk-for-go/pull/64/files#r26442000>.